src/test/java/bit/javaoop/CompanyTest.java

```diff
@@ -0,0 +1,69 @@
+package bit.javaoop;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class CompanyTest {
+
+    @Test
+    void toString_1() {
+        AbstractEmployee employee = new Employee("Wacek", 8_000);
+        AbstractEmployee employee3 = new Employee("Leszek", 7_000);
+        AbstractEmployee employee4 = new Employee("Frodo", 7_000);
+        AbstractManager manager1 = new FixedBudgetManager("Grzesiek", 25_000, 100_000);
+        AbstractManager manager2 = new FixedBudgetManager("Maciek", 25_000, 100_000);
+        AbstractManager manager3 = new FixedEmployeeNumberManager("Krzysiek", 19_000, 2);
+        CEO ceo_1 = new CEO("Zbyszek");
+        Company company = new Company("THICC");
+        company.hireNewCeo(ceo_1);
+        ceo_1.addManager(manager2);
+        ceo_1.addManager(manager1);
+        ceo_1.addManager(manager3);
+        manager2.hire(employee);
+        manager2.hire(employee3);
+        manager3.hire(employee4);
+        String output = company.toString();
+        assertTrue(output.equals("THICC - Company name\n" +
+                "\tZbyszek - CEO\n" +
+                "\t\tMaciek - Manager\n" +
+                "\t\t\tWacek - Employee\n" +
+                "\t\t\tLeszek - Employee\n" +
+                "\t\tGrzesiek - Manager\n" +
+                "\t\tKrzysiek - Manager\n" +
+                "\t\t\tFrodo - Employee\n"));
+    }
+
+    @Test
+    void newCeo() {
+        AbstractEmployee employee = new Employee("Wacek", 8_000);
+        AbstractEmployee employee3 = new Employee("Leszek", 7_000);
+        AbstractEmployee employee4 = new Employee("Frodo", 7_000);
+        AbstractManager manager1 = new FixedBudgetManager("Grzesiek", 25_000, 100_000);
+        AbstractManager manager2 = new FixedBudgetManager("Maciek", 25_000, 100_000);
+        AbstractManager manager3 = new FixedEmployeeNumberManager("Krzysiek", 19_000, 2);
+        CEO ceo_1 = new CEO("Zbyszek");
+        CEO ceo_2 = new CEO("Walery");
+        Company company = new Company("THICC");
+        company.hireNewCeo(ceo_1);
+        ceo_1.addManager(manager2);
+        ceo_1.addManager(manager1);
+        ceo_1.addManager(manager3);
+        manager2.hire(employee);
+        manager2.hire(employee3);
+        manager3.hire(employee4);
+        company.hireNewCeo(ceo_2);
+        manager3.hire(employee);
+        String output = company.toString();
+        assertTrue(output.equals("THICC - Company name\n" +
+                "\tWalery - CEO\n" +
+                "\t\tMaciek - Manager\n" +
+                "\t\t\tWacek - Employee\n" +
+                "\t\t\tLeszek - Employee\n" +
+                "\t\tGrzesiek - Manager\n" +
+                "\t\tKrzysiek - Manager\n" +
+                "\t\t\tFrodo - Employee\n" +
+                "\t\t\tWacek - Employee\n"));
+    }
+
+}
```

Comment:
Wyekstrahowałbym cały podstawowy setup do metody setUp oznaczonej adnotacją @BeforeEach, która sprawi, że ta metoda będzie wołana przed każdym naszym testem.
w ten sposób:
![image](https://user-images.githubusercontent.com/18233796/37921006-d86710a8-3128-11e8-93e8-1ed7fa724c28.png)
generalnie jeśli kiedykolwiek widzisz, że w klasie kod powtarza Ci się choćby w jednym miejscu to znak, że prawdopodobnie warto wyekstrahować ten kawałek do osobnej metody prywatnej (ctrl + alt + m).

Oprócz tego polecam następującą konwencję nazewnictwa testów - odrazu widać co metoda testuje "shouldCosZrobic"
+ szablon //given //when //then

Ogólnie zasada jest taka jedna metoda powinna robić dokładnie jedną rzecz tak samo jest z testami.
W drugim teście testujesz zmianę CEO + jakieś dodatkowe dopisywanie pracowników po zmianie powinno to być przetestowane osobno.
Po wprowadzeniu zmian u mnie metody wyglądają następująco:
![image](https://user-images.githubusercontent.com/18233796/37921212-72239a5e-3129-11e8-92c0-89842e1042da.png)

![image](https://user-images.githubusercontent.com/18233796/37921205-6b864eda-3129-11e8-8656-5569abb26b28.png)

Prawda, że ładniej:)?